File: Baekjoon/Implementation/[2776]암기왕/solution.py
import sys

T = int(sys.stdin.readline())

while T != 0:
    N = int(sys.stdin.readline())
    num = list(map(int, sys.stdin.readline().split()))
    note = dict()

    for n in num:
        note[n] = True

    M = int(sys.stdin.readline())
    num = list(map(int, sys.stdin.readline().split()))

    for n in num:
        if note.get(n):
            sys.stdout.write("1\n")
        else:
            sys.stdout.write("0\n")

    T -= 1

File: Baekjoon/Implementation/[2776]암기왕/solution.js
const fileName = process.platform === "linux" ? "/dev/stdin" : "example.dat";
const inputData = require("fs").readFileSync(fileName).toString().split("\n");

const input = (() => {
  let line = 0;
  return () => inputData[line++];
})();

const T = Number(input());

for (let t = 0; t < T; t++) {
  const N = Number(input());
  const note = new Set();
  input()
    .split(" ")
    .map((item) => note.add(item));
  const M = Number(input());
  console.log(
    input()
      .split(" ")
      .map((item) => (note.has(item) ? 1 : 0))
      .join("\n")
  );
}
